integrationTests/mock/oneSCExecutorMockVM.go

```diff
@@ -218,6 +222,55 @@ func (vm *OneSCExecutorMockVM) processAddFunc(input *vmcommon.ContractCallInput,
 	}, nil
 }
 
+func (vm *OneSCExecutorMockVM) processWithdrawFunc(input *vmcommon.ContractCallInput, value *big.Int) (*vmcommon.VMOutput, error) {
+
+	destNonce, err := vm.blockchainHook.GetNonce(input.RecipientAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	destBalance, err := vm.blockchainHook.GetBalance(input.RecipientAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	if destBalance.Cmp(value) < 0 {
+		return nil, errInsuficientFunds
+	}
+
+	newSCBalance := big.NewInt(0).Sub(destBalance, value)
```

Comment:
The processAdd method has here the input.CallerValue, why?

Reply:
Because if you deposit you send a transaction with a `Value`. If you withdraw, I assumed you make a call to the Withdraw function in the smart contract, which has an argument. The `Value` field will be 0 on the tx (in order not to be considered a payment) and the argument will be the amount you want to take out.